content/blog/building-images-bake/index.md

```diff
@@ -0,0 +1,169 @@
+`localhost:5000`, which is the default registry defined for testing environments in the parent Bake file. Let's explain the full command:
+
+As is defined in the [Bake documentation about remote definitions](https://docs.docker.com/build/bake/remote-definition/)
+you can use a remote Bake definition with all the functions and default targets, and attach another local one that you can use to override
+all the default values.
+In the command above, `-f cwd://bake.hcl` is the local file that we created in Step 1, and
+`-f docker-bake.hcl` is the remote file in the git repo, that we're using to build the image.
+
+You can explore more about all the content generated and used inside the Bake file by appending the `--print` flag, as in the following command:
+
+```bash
+docker buildx bake -f docker-bake.hcl -f cwd://bake.hcl "https://github.com/cloudnative-pg/postgres-containers.git" myimage --print
+```
+
+### Step 3: Push the image to a registry
+
+Now you just need to push the image to a registry. You can do this by using the following command:
+
+```bash
+registry=your/registry:5000 docker buildx bake -f docker-bake.hcl -f cwd://bake.hcl "https://github.com/cloudnative-pg/postgres-containers.git" myimage --push
+```
+
+The previous command will push the images in the following format: `your/registry:5000/postgresql-testing:17-standard-bookworm`.
+Using the `--print` flag you can explore the full list of tags created that are in the parent Bake file.
+
+### Step 4: Serve the image
+
+You can now let your clusters use the image that we've built based on the CloudNativePG operand images provided
+by the CloudNativePG project.
+
+## Deep dive into the Bake and Dockerfile
+
+The simplicity of Bake to do even more stuff is amazing, and allows you to create custom images easily.
+But, how does this magic happen? Let's take a look at the Bake and the Docker file.
+
+### Bake file
+
+The magic starts with our [postgres-containers repository](https://github.com/cloudnative-pg/postgres-containers),
+where we have a `docker-bake.hcl` file that is being used to build the images provided by the CloudNativePG project.
+It's the base for our custom Bake file.
+
+The `docker-bake.hcl` file contains a lot of functions that are used to build the images. One of them is the `getExtensionsString()`.
+This function, given the list of extensions we provided, will return a string of the extensions with the correct package name
+for a Debian-based distribution, in our case, Debian Bookworm.
+For example, the `pg_vector` extension will be translated into
+`postgresql-16-pgvector,` which is the name of the package for pgvector extensions for PostgreSQL 16 in the Debian
+Bookworm distribution.
+
+When we add elements to, for example, the `args` variable, those elements are processed by the Bake file, and will be
+merged, meaning that the new elements will be added, and the existing ones will be overwritten.
```

Comment:
"processed by the Bake file"??
Or processed by the Docker bake command?

And, do we need to explain what merge means?
"meaning that the new elements will be added, and the existing ones will be overwritten."